spacy/pipeline/entity_linker.py

```diff
@@ -453,126 +445,129 @@ def predict(self, docs: Iterable[Doc]) -> ActivationsT:
             }
         if isinstance(docs, Doc):
             docs = [docs]
-        for doc in docs:
+
+        docs = list(docs)
+        # Determine which entities are to be ignored due to labels_discard.
+        valid_ent_idx_per_doc = (
+            [
+                idx
+                for idx in range(len(doc.ents))
+                if doc.ents[idx].label_ not in self.labels_discard
+            ]
+            for doc in docs
+            if len(doc) and len(doc.ents)
+        )
+
+        # Call candidate generator.
+        all_ent_cands = self.get_candidates(
+            self.kb,
+            (
+                SpanGroup(
+                    doc,
+                    spans=[doc.ents[idx] for idx in next(valid_ent_idx_per_doc)],
+                )
+                for doc in docs
+                if len(doc) and len(doc.ents)
+            ),
+        )
```

Comment:
It's not clear to me why we need `valid_ent_idx_per_doc` and we can't just call something like this directly:

``` 
spans =  [ent in doc.ents if ent.label_ not in self.labels_discard]
```

Reply:
Don't remember why I wrote it like this. Originally it used in two different spots, I think. Anyway, updated in https://github.com/explosion/spaCy/pull/12367/commits/ee5d7f4a3341f350bcb20d888b609601ce3ca086.